src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -152,7 +152,11 @@ internal abstract partial class AbstractProject : ForegroundThreadAffinitizedObj
 
             if (visualStudioWorkspaceOpt != null)
             {
-                this.EditAndContinueImplOpt = new VsENCRebuildableProjectImpl(this);
+                if (Language == LanguageNames.CSharp || Language == LanguageNames.VisualBasic)
```

Comment:
I briefly looked into doing so but CPSProject doesn't seem to have specialized subclasses for C# and VB. So the language check would need to be there in addition to having virtual property.